collector/load_balancer_collector_test.go

```diff
@@ -0,0 +1,234 @@
+package collector
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/vmware/go-vmware-nsxt/loadbalancer"
+)
+
+const (
+	fakeLoadBalancerID             = "fake-load-balancer-id"
+	fakeLoadBalancerName           = "fake-load-balancer-name"
+	fakeLoadBalancerPoolID         = "fake-load-balancer-pool-id"
+	fakeLoadbalancerPoolMemberIP   = "127.0.0.1"
+	fakeLoadbalancerPoolMemberPort = "9732"
+)
+
+type mockLoadBalancerClient struct {
+	responses []mockLoadBalancerResponse
+}
+
+type mockLoadBalancerResponse struct {
+	ID               string
+	Name             string
+	Status           string
+	PoolID           string
+	PoolStatus       string
+	PoolMemberStatus string
+	Error            error
+}
+
+func (c *mockLoadBalancerClient) ListAllLoadBalancers() ([]loadbalancer.LbService, error) {
+	panic("unused function. Only used to satisfy LoadBalancerClient interface")
+}
+
+func (c *mockLoadBalancerClient) GetLoadBalancerStatus(loadBalancerID string) (loadbalancer.LbServiceStatus, error) {
+	for _, res := range c.responses {
+		if res.ID == loadBalancerID {
+			return loadbalancer.LbServiceStatus{
+				ServiceId:     res.ID,
+				ServiceStatus: res.Status,
+				Pools: []loadbalancer.LbPoolStatus{
+					{
+						PoolId: res.PoolID,
+						Status: res.PoolStatus,
+						Members: []loadbalancer.LbPoolMemberStatus{
+							{
+								IPAddress: fakeLoadbalancerPoolMemberIP,
+								Port:      fakeLoadbalancerPoolMemberPort,
+								Status:    res.PoolMemberStatus,
+							},
+						},
+					},
+				},
+			}, res.Error
+		}
+	}
+	return loadbalancer.LbServiceStatus{}, errors.New("load balancer not found")
+}
+
+func buildLoadBalancerStatusResponse(id string, status string, poolStatus string, poolMemberStatus string, err error) mockLoadBalancerResponse {
+	return mockLoadBalancerResponse{
+		ID:               fmt.Sprintf("%s-%s", fakeLoadBalancerID, id),
+		Name:             fmt.Sprintf("%s-%s", fakeLoadBalancerName, id),
+		Status:           status,
+		PoolID:           fmt.Sprintf("%s-%s", fakeLoadBalancerPoolID, id),
+		PoolStatus:       poolStatus,
+		PoolMemberStatus: poolMemberStatus,
+		Error:            err,
+	}
+}
+
+func TestLoadBalancerCollector_GenerateLoadBalancerStatusMetrics(t *testing.T) {
+	testcases := []struct {
+		description           string
+		loadBalancerResponses []mockLoadBalancerResponse
+		expectedMetrics       []loadBalancerStatusMetric
+	}{
+		{
+			description: "",
```

Comment:
intentionally left as empty? if description is not used we can just remove it from the struct

Reply:
Oops, forgot to add a description and other testcases